packages/framework/src/hooks/useStyleNames.ts

```diff
@@ -1,14 +1,60 @@
 import { useAtomValue } from "jotai";
 import type { CSSProperties } from "react";
-import { type Expression, resolveStyleNames } from "../shared";
-import { themeAtom } from "../state";
-import { isString } from "lodash-es";
+import { get, intersection, isEmpty, isString, keys, merge } from "lodash-es";
+import {
+  type EnsembleThemeModel,
+  isExpression,
+  type Expression,
+} from "../shared";
+import { themeAtom, defaultScreenContext } from "../state";
+import { evaluate } from "../evaluate";
+import { type EnsembleStorage, useEnsembleStorage } from "./useEnsembleStorage";
+import { type CustomScope, useCustomScope } from "./useCustomScope";
+
+const resolveStyleNames = (
+  styleNames: string[] | string,
+  theme: EnsembleThemeModel,
+  storage: EnsembleStorage,
+  customScope?: CustomScope,
+): CSSProperties => {
+  let names: string[] = isString(styleNames)
+    ? styleNames.split(" ")
+    : styleNames;
+
+  names = names.map((styleName) => {
+    const isClass = styleName.startsWith(".");
+    const styleKey = isClass ? styleName.substring(1) : styleName;
+    const styleVal = isExpression(styleKey)
+      ? evaluate<string>(defaultScreenContext, styleKey, {
```

Comment:
Don't use `evaluate` directly. Did you see the warning? Use `useEvaluate` and then update the dependency array of the second evaluation to include styles like we discussed.